cmd/cluster-agent/api/server.go

```diff
@@ -123,7 +123,7 @@ func StartServer(w workloadmeta.Component, senderManager sender.DiagnoseSenderMa
 
 	grpcSrv := grpc.NewServer(opts...)
 	pb.RegisterAgentSecureServer(grpcSrv, &serverSecure{
-		taggerServer: taggerserver.NewServer(tagger.GetDefaultTagger()),
+		taggerServer: taggerserver.NewServer(tagger.GetTaggerInstance()),
```

Comment:
dummy question: why we don't provide the tagger as parameter of this function? 

```golang
// current
func StartServer(w workloadmeta.Component, senderManager sender.DiagnoseSenderManager) error { //..}

// propose
func StartServer(w workloadmeta.Component, taggerCmp tagger.Component, senderManager sender.DiagnoseSenderManager) error { //..}
```

Is it is something plane in a follow up PR? is it not possible?